### PR TITLE
Fix sanitization of DC label (fixes #622)

### LINKED ISSUE #622: Fix sanitization of the datacenter label
When creating a DC with case-sensitive CQL name overrides:
```yaml
apiVersion: cassandra.datastax.com/v1beta1
kind: CassandraDatacenter
metadata:
  namespace: test
  name: k8s-dc1
spec:
  clusterName: Cql_Cluster
  datacenterName: Cql_DataCenter1
  racks:
    - name: Cql_Rack1
  serverType: cassandra
  serverVersion: 4.1.0
  size: 1
  storageConfig:
    cassandraDataVolumeClaimSpec:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 100Mi
      storageClassName: standard
```
The following labels are applied to the Cassandra pods:
```
cassandra.datastax.com/cluster=Cql_Cluster
cassandra.datastax.com/datacenter=cql-datacenter1
cassandra.datastax.com/rack=Cql_Rack1
```
The datacenter label is inconsistent: it should be sanitized with `CleanLabelValue` instead of `CleanupForKubernetes`.

### PR DESCRIPTION
**Pull request by olim7t**
**What this PR does**:
Fix sanitization of DC label

**Which issue(s) this PR fixes**:
Fixes #622

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [ ] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: pkg/reconciliation/reconcile_datacenter.go [hunk 1]
```diff
@@ -144,7 +144,7 @@ func (rc *ReconciliationContext) listPVCs() (*corev1.PersistentVolumeClaimList,
 	rc.ReqLogger.Info("reconciler::listPVCs")
 
 	selector := map[string]string{
-		api.DatacenterLabel: rc.Datacenter.Name,
+		api.DatacenterLabel: api.CleanLabelValue(rc.Datacenter.DatacenterName()),
```
Comment (olim7t) [inline pkg/reconciliation/reconcile_datacenter.go:147]: This was probably a bug.

File: tests/test_all_the_things/test_all_the_things_suite_test.go [hunk 2]
```diff
@@ -133,7 +133,7 @@ var _ = Describe(testName, func() {
 			wg.Add(1)
 			go func() {
 				k = kubectl.Logs("-f").
-					WithLabel(fmt.Sprintf("statefulset.kubernetes.io/pod-name=cluster1-%s-r1-sts-0", api.CleanupForKubernetes(dcNameOverride))).
+					WithLabel(fmt.Sprintf("statefulset.kubernetes.io/pod-name=cluster1-%s-r1-sts-0", api.CleanLabelValue(dcNameOverride))).
```
Comment (burmanm) [inline tests/test_all_the_things/test_all_the_things_suite_test.go]: issue: This is wrong and the test shouldn't pass if we mess up naming of the StatefulSet. The StatefulSet name must adhere to the CleanupForKubernetes process (minimum), not CleanLabelValue as those values are not allowed in the StatefulSet's name.

Although we're polling for label value here, the DNS name of the pod matters. The name defined must match what's in here:

https://github.com/k8ssandra/cass-operator/blob/master/pkg/reconciliation/construct_statefulset.go#L29

Comment (olim7t) [reply tests/test_all_the_things/test_all_the_things_suite_test.go]: Ah, my bad, I saw a label and just changed the function, not realizing that the value was actually referencing a resource name. I'll revert this.

**Comment by olim7t** [comment]
@burmanm is the `decommmission_dc` integration test flaky? I see it's failing on master too.

**Review by adejanovski — APPROVED** [review]

**Pull request merged**